adapters.d.ts

```diff
@@ -43,47 +43,47 @@ export interface AdapterInstance<
   TSession,
   TVerificationRequest
 > {
-  createUser(profile: TProfile): Promise<TUser>
-  getUser(id: string): Promise<TUser | null>
-  getUserByEmail(email: string): Promise<TUser | null>
-  getUserByProviderAccountId(
+  createUser: (profile: TProfile) => Promise<TUser>
```

Comment:
I disagree with this change strongly. It changes autocompletion. It makes these appear as properties instead of methods (the icon). The problem is that properties can be callable or not. It just makes the autocomplete less usable, because it is harder to tease apart callables from non-callables.